c/include/libsbp/unpacked/string/multipart.h

```diff
@@ -0,0 +1,68 @@
+#ifndef LIBSBP_UNPACKED_STRING_MULTIPART_H
+#define LIBSBP_UNPACKED_STRING_MULTIPART_H
+
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <libsbp/common.h>
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#define SBP_MULTIPART_STRING_MAX_SECTIONS 20
```

Comment:
Where did the max of 20 come from?

Reply:
Chosen at random. This can be removed at the expense of CPU time. I went through several different versions of the string functions and they were all pretty terrible, but they proved a point. I think they all need to be rewritten and, like you said elsewhere, consolidate.

Reply:
Ok, 20 seems like a reasonable upper limit given the max usage that I know of is 3 for the settings code

Reply:
There are a few messages which can potentially have more. settings read resp (I think) has an optional 4th section which is only present for enum types. Some of the more piksi specific messages like read_dir_resp or process_fd_summary are technically unbounded, although there is practical limit based on the length of filenames and such.